src/main/java/com/cascadebot/cascadebot/commands/moderation/BanCommand.java

```diff
@@ -30,17 +32,46 @@ public void onCommand(Member sender, CommandContext context) {
         }
 
         Member targetMember = DiscordUtils.getMember(context.getGuild(), context.getArg(0));
+        User targetUser;
+        String reason = null;
+
+        if (context.getArgs().length >= 2) {
+            reason = context.getMessage(1);
+        }
 
         if (targetMember == null) {
-            context.replyDanger(MessagingObjects.getStandardMessageEmbed("We couldn't find that user in this guild!\n" +
-                    "To forcibly ban a user not in this guild, use `;forceban`!", sender.getUser()));
-            return;
+            targetUser = DiscordUtils.getUser(context.getGuild(), context.getMessage(0), true);
+        } else {
+            targetUser = targetMember.getUser();
         }
 
-        String reason = null;
+        if (targetUser == null) {
+            context.replyDanger("We could not find that user!");
+            return;
+        }
 
-        if (context.getArgs().length >= 2) {
-            reason = context.getMessage(1);
+        if (targetMember == null) {
+            String finalReason = reason;
+            ConfirmUtils.confirmAction(
```

Comment:
We should first check to make sure the User actually exists before doing confirm utils. in fact if a user is found this won't work at all. If a user isn't found send the user not found message.

Reply:
We do this on line 51